arduino-ide-extension/src/common/protocol/boards-service.ts

```diff
@@ -553,6 +554,22 @@ export namespace Board {
     return left.name === right.name && left.fqbn === right.fqbn;
   }
 
+  export function sameDistinctHardwareAs(
+    left: Board,
+    right: string | Board
```

Comment:
A `string` type does not make sense here because it will always be falsy. It was copied based on `Board#sameAs` where it can compare a `string` with a `Board`. Please filter on the caller side and do not introduce such unused APIs.

Reply:
good catch, removed `string`, at present callers are always passing a `Board`